### PR TITLE
Throw an error if we get something on stderr

### PR DESCRIPTION
**Pull request by steelbrain**
when we were expecting it to be on stdout
/cc @joefitzgerald @AtomLinter/linter

### CODE REVIEW

**Comment by steelbrain** [comment]
I have a spec planned for this but I am a little short on time

File: lib/helpers.coffee [hunk 1]
```diff
@@ -25,18 +25,22 @@ module.exports = Helpers =
...
         reject(err)
       spawnedProcess.on 'close', ->
-        resolve(data.join(''))
+        if options.stream is 'stdout'
+          if data.stderr.length
+            reject(data.stderr.join(''))
+          else
+            resolve(data.stdout.join(''))
+        else
+          resolve(data.stderr.join(''))
```
Comment (park9140) [inline lib/helpers.coffee]: we resolve with stderr here?  Why?

Comment (steelbrain) [inline lib/helpers.coffee]: Because we are expecting stderr, `options.stream` is stderr :wink:

Comment (park9140) [inline lib/helpers.coffee]: Oh right... because linters on the command line often use stderr to report results when they fail.

**Comment by park9140** [comment]
:ship:

**Pull request merged**